crates/sui-core/src/narwhal_manager/mod.rs

```diff
@@ -166,8 +167,11 @@ where
         execution_state,
         &registry,
     )
-    .await
-    .unwrap();
+    .await;
+    match primary_result {
+        Ok(n) => narwhal_handles = n,
+        Err(e) => panic!("Unable to start Narwhal Primary {:?}", e),
+    }
```

Comment:
I don't know if that's the intention, but the `panic` raised in primary when retries have depleted won't be caught here - other primary should return a `Result` instead as well

Reply:
Good point. This is also the case for all the other panics. We can think about either handling all possible transient errors in place, or, always returning errors instead of raising panic, and having the upstream handle all possible transient errors.